MDANSE/Src/MDANSE/Framework/Configurators/GroupingLevelConfigurator.py

```diff
@@ -72,18 +81,59 @@ def configure(self, value):
 
         SingleChoiceConfigurator.configure(self, value)
 
-        if value == "atom":
-            return
+        self["level"] = value
+
         trajConfig = self._configurable[self._dependencies["trajectory"]]
         atomSelectionConfig = self._configurable[self._dependencies["atom_selection"]]
         chemical_system = trajConfig["instance"].chemical_system
+
+        if value == "atom" or value == "each atom":
+            return
+
         indices = []
         elements = []
         names = []
         masses = []
+        group_names = []
+        group_elements = {}
+        group_n_atms = {}
         mass_lookup = chemical_system.atom_property("atomic_weight")
 
         if value == "molecule":
+            if len(trajConfig["instance"].chemical_system.unique_molecules()) == 0:
+                self.error_status = "The trajectory does not contain molecules."
+                return
+
+            for mol_name in chemical_system._clusters.keys():
```

Comment:
If a `dict` `keys` is default iterator.
```suggestion
            for mol_name in chemical_system._clusters:
```